src/SoapClient.php

```diff
@@ -117,38 +123,51 @@ class SoapClient
     public function __construct(SoapFactory $factory = null)
     {
         $this->factory = $factory;
-        $this->setHandler();
-        $this->wsdlProvider = LocalWsdlProvider::create();
+        $this->client = new Client($this->guzzleClientOptions);
+        $this->pluginClient = new PluginClient($this->client, $this->middlewares);
+        $this->wsdlProvider = Psr18Loader::createForClient($this->pluginClient);
```

Comment:
Maybe best to wrap the FlatteningLoader here (for imports and includes)

Reply:
Done